other/best_match_finder.py

```diff
@@ -0,0 +1,129 @@
+def get_best_match(
+    group1: dict[object, list], group2: dict[object, list]
+) -> list[tuple]:
+    """
+    Match elements from group1 and group2 based on their preferences,
+    such that there are no rogue couples (Two elements such that they
+    have a higher preference for each other than the element they
+    have been matched with)
+
+    This algorithm comes from the MITOCW lecture from the course 6.042
+    by Prof. Tom Leighton
+
+    The algorithm discussed with an example:
+    https://youtu.be/5RSMLgy06Ew?list=PL6MpDZWD2gTF3mz26HSufmsIO-COKKb5j&t=1623
+
+
+    :param group1: Preferences of elements of group1 as a dict of the form
+    {<group1_elemt>: [...<List of group2_elemts in descending order of preference>]}
+    :param group2: Preferences of elements of group2 as a dict of the form
+    {<group2_elemt>: [...<List of group1_elemts in descending order of preference>]}
+    """
+    group1_checklist: dict[object, list[object]] = dict(
+        zip(group1.keys(), list(group1.values()).copy())
+    )
+    group2_possible_matches: dict[object, list[object]] = dict(
+        zip(group2.keys(), [[]] * 5)
+    )
+
```

Comment:
You should typehint the return type of this function

Reply:
Could you help me on how to typehint this one? This one is actually an assignment.

Reply:
Yes, I added the typehints for all functions now and took the nested functions out of the `get_best_match` function for more readability